Src/PerformanceCollector/Filtering/CollectionConfiguration.cs

```diff
@@ -0,0 +1,331 @@
+
+        private readonly List<DocumentStream> documentStreams = new List<DocumentStream>();
+
+        private readonly List<Tuple<string, string>> performanceCounters = new List<Tuple<string, string>>();
+
+        public CollectionConfiguration(
+           CollectionConfigurationInfo info,
+           out CollectionConfigurationError[] errors,
+           Clock timeProvider,
+           IEnumerable<DocumentStream> previousDocumentStreams = null)
+        {
+            if (info == null)
+            {
+                throw new ArgumentNullException(nameof(info));
+            }
+
+            this.info = info;
+
+            // create metrics based on descriptions in info
+            CollectionConfigurationError[] metricErrors;
+            this.CreateMetrics(info, out metricErrors);
+
+            // maintain a separate collection of all (Id, AggregationType) pairs with some additional data - to allow for uniform access to all types of metrics
+            this.CreateMetadata();
+
+            // create document streams based on description in info
+            CollectionConfigurationError[] documentStreamErrors;
+            this.CreateDocumentStreams(out documentStreamErrors, timeProvider, previousDocumentStreams ?? new DocumentStream[0]);
+
+            // check performance counters
+            CollectionConfigurationError[] performanceCounterErrors;
+            this.CreatePerformanceCounters(out performanceCounterErrors);
+
+            errors = metricErrors.Concat(documentStreamErrors).Concat(performanceCounterErrors).ToArray();
+
+            foreach (var error in errors)
+            {
+                error.Data["ETag"] = this.info.ETag;
+            }
+        }
+
+        public IEnumerable<OperationalizedMetric<RequestTelemetry>> RequestMetrics => this.requestTelemetryMetrics;
+
+        public IEnumerable<OperationalizedMetric<DependencyTelemetry>> DependencyMetrics => this.dependencyTelemetryMetrics;
+
+        public IEnumerable<OperationalizedMetric<ExceptionTelemetry>> ExceptionMetrics => this.exceptionTelemetryMetrics;
+
+        public IEnumerable<OperationalizedMetric<EventTelemetry>> EventMetrics => this.eventTelemetryMetrics;
+
+        public IEnumerable<OperationalizedMetric<TraceTelemetry>> TraceMetrics => this.traceTelemetryMetrics;
+
+        public IEnumerable<OperationalizedMetric<MetricValue>> MetricMetrics => this.metricMetrics;
+
+        /// <summary>
+        /// Telemetry types only (handled by QuickPulseTelemetryProcessor).
+        /// </summary>
+        public IEnumerable<Tuple<string, AggregationType>> TelemetryMetadata => this.telemetryMetadata;
+
+        /// <summary>
+        /// Metric type only (handled by QuickPulseMetricProcessor).
+        /// </summary>
+        public IEnumerable<Tuple<string, AggregationType>> MetricMetadata => this.metricMetadata;
+
+        /// <summary>
+        /// Document streams (handled by QuickPulseTelemetryProcessor).
```

Comment:
It means that this collection is used from QuickPulseTelemetryProcessor at collection-time. I've revised the comment.